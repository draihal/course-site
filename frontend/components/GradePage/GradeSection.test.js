import React from 'react'
import renderer from 'react-test-renderer'
import { fireEvent, render } from '@testing-library/react'
import TestGradeSection from './GradeSection'

const homework1 = {
  'id': 1,
  'lesson': 111,
  'student': 11,
  'student_first_name': 'Testing First Name',
  'student_last_name': 'Testing Last Name',
  'student_homework': 'Testing Homework',
  'grade': {
    'status': 'rework',
    'grade': 'Test grade',
    'lesson': 2,
    'id': 1,
    'student': 22
  }
};

const homework2 = {
  'id': 1,
  'lesson': 111,
  'student': 11,
  'student_first_name': 'Testing First Name 2',
  'student_last_name': 'Testing Last Name',
  'student_homework': 'Testing Homework',
  'grade': []
};


describe('GradeSection', () => {
  it('render correctly GradeSection component', () => {
    const tree = renderer.create(
      <TestGradeSection teacherId={'13'} homework={homework1} token={null} />
    ).toJSON();
    expect(tree).toMatchSnapshot()
  });

  it('render GradeSection correctly with given props with grade', () => {
    const { queryByText } = render(
      <TestGradeSection teacherId={'13'} homework={homework1} token={null} />
    );
    expect(queryByText(/Testing First Name/i)).toBeTruthy();
    expect(queryByText(/Test grade/i)).toBeTruthy();
  });

  it('render GradeSection correctly with given props without grade', () => {
    const { queryByText } = render(
      <TestGradeSection teacherId={'13'} homework={homework2} token={null} />
    );
    expect(queryByText(/Testing First Name 2/i)).toBeTruthy();
    expect(queryByText(/Test grade/i)).toBeFalsy();
  });

});