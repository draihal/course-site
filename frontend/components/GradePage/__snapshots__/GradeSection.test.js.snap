// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GradeSection render correctly GradeSection component 1`] = `
<div>
  <div
    className="group py-4"
  >
    <div
      className="container"
    >
      <h2>
        Домашнее задание от 
        Testing First Name
         
        Testing Last Name
      </h2>
      <hr />
      <form
        className="form-horizontal form-profile"
        onSubmit={[Function]}
        role="form"
      >
        <div
          className="form-group row"
        >
          <label
            className="col-lg-2 control-label"
          >
            Ответ студента
          </label>
          <div
            className="col-lg-10"
          >
            <input
              className="form-control-plaintext"
              readOnly={true}
              type="text"
              value="Testing Homework"
            />
          </div>
        </div>
        <div
          className="form-group row"
        >
          <label
            className="col-lg-2 control-label"
            htmlFor="grade_status"
          >
            Статус проверки
          </label>
          <div
            className="col-lg-10"
          >
            <div
              className="ui-select"
            >
              <select
                className="form-control"
                id="grade_status"
                onChange={[Function]}
                value="rework"
              >
                <option
                  value="undone"
                >
                  Не сдано
                </option>
                <option
                  value="check"
                >
                  На проверке
                </option>
                <option
                  value="rework"
                >
                  На доработке
                </option>
                <option
                  value="done"
                >
                  Сдано
                </option>
              </select>
            </div>
          </div>
        </div>
        <div
          className="form-group row"
        >
          <label
            className="col-lg-2 control-label"
            htmlFor="grade_textarea"
          >
            Оценка
          </label>
          <textarea
            className="col-lg-10 form-control"
            id="grade_textarea"
            onChange={[Function]}
            required={true}
            rows="4"
            value="Test grade"
          />
        </div>
        <button
          className="btn btn-secondary btn-md float-right"
          type="submit"
        >
          Изменить
        </button>
      </form>
    </div>
  </div>
</div>
`;
